.github/workflows/java_thin_client_compatibility.yaml

```diff
@@ -0,0 +1,117 @@
+        run: |
+          versions=$( python get_server_matrix.py )
+          echo "Versions from Python script: $versions"
+          filtered_versions=$(echo "$versions" | jq -c '[.[] | select(. > "5.2")]')
+          echo "Filtered versions: $filtered_versions"
+          echo "matrix=$filtered_versions" >> "$GITHUB_OUTPUT"
+
+  test_client:
+    needs: [setup_server_matrix]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        server-version: ${{ fromJson(needs.setup_server_matrix.outputs.matrix) }}
+        kind: [os, enterprise]
+        os: [ ubuntu-latest ]
+        java: [ { version: 17, distribution: corretto },
+                { version: 21, distribution: corretto },
+                { version: 17, distribution: zulu },
+                { version: 21, distribution: zulu },
+                { version: 17, distribution: semeru },
+                { version: 21, distribution: semeru },
+                { version: 17, distribution: oracle },
+                { version: 21, distribution: oracle },
+                { version: 17, distribution: microsoft },
+                { version: 21, distribution: microsoft }
+              ]
+
+    name: Test Java Thin client ${{ github.event.inputs.branch_name }} branch against ${{ matrix.kind }} ${{ matrix.server-version }} server on ${{ matrix.os }} and JVM ${{ matrix.java.version }} ${{ matrix.java.distribution }}
+    steps:
+      - name: Checkout to scripts
+        uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ matrix.java.version }}
+          distribution: ${{ matrix.java.distribution }}
+
+      - name: Checkout to ${{ github.event.inputs.branch_name }}
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.inputs.organization_name }}/hazelcast-java-client
+          token: ${{ secrets.GH_PAT }}
+          path: client
+          ref: ${{ github.event.inputs.branch_name }}
+
+      - name: Setup Local Maven Cache
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Check Server Version to Determine License Version
+        uses: madhead/semver-utils@latest
+        id: version
+        with:
+          version: ${{ matrix.server-version }}
+          compare-to: 5.3.0
+
+      - name: Set up HZ_LICENSEKEY env
+        if: ${{ matrix.kind == 'enterprise' }}
+        run: |
+          echo "HZ_LICENSEKEY=${{ '>' == steps.version.outputs.comparison-result && secrets.HAZELCAST_ENTERPRISE_KEY || secrets.HAZELCAST_ENTERPRISE_KEY_V5 }}" >> $GITHUB_ENV
+
+      - name: Set up HAZELCAST_VERSION env
+        shell: bash -l {0}
+        run: |
+          echo "HAZELCAST_VERSION=$( mvn help:evaluate -Dexpression=project.version -q -DforceStdout )" >> $GITHUB_ENV
```

Comment:
matrix.server-version should already have the hazelcast version. Why do we have this step? Also can you remind me what HAZELCAST_VERSION is used for?

Reply:
This is a copy-paste. I will remove this.

Reply:
Fixed